cmd/node.go

```diff
@@ -251,7 +296,43 @@ func (node *Node) Start(exitChannels ...*chan struct{}) {
 			}
 		}
 	}
-	node.IsRunning = true
+
+	node.statusMutex.Lock()
+	// Load the node status.
+	// This is identify whether the node is initialized for the first time or it's a restart.
+	status, err := node.LoadStatus()
```

Comment:
same as I mentioned below about moving mutex usage into these helper function. Can we move the lock/unlock logic into `node.LoadStatus` and `node.UpdateStatusRunning`?